bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/MetOfficeDataHubSiteApiHandler.java

```diff
@@ -0,0 +1,839 @@
+        MetOfficeDataHubBridgeHandler uplinkBridge = getMetOfficeDataHubBridge();
+        if (uplinkBridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    getLocalizedText("site.error.no-bridge"));
+            return;
+        }
+
+        final boolean authFailedPreviously = authFailed;
+
+        if (!getThing().getStatus().equals(ThingStatus.ONLINE) && !authFailedPreviously) {
+            logger.debug("Disabled requesting data - this thing is not ONLINE");
+            return;
+        }
+
+        if (!authFailed
+                && uplinkBridge.forecastDataLimiter.getRequestCountIfAvailable() == RequestLimiter.INVALID_REQUEST_ID) {
+            logger.debug("Disabled requesting data - request limit has been hit");
+            return;
+        } else {
+            uplinkBridge.updateLimiterStats();
+        }
+
+        authFailed = false;
+
+        final Response.CompleteListener listener = new BufferingResponseListener() { // 4.5kb buffer will cover both
+                                                                                     // requests
+            @Override
+            public void onComplete(@Nullable Result result) {
+                if (result != null && !result.isFailed()) {
+                    final String response = getContentAsString();
+                    if (response != null) {
+                        scheduler.execute(() -> {
+                            logger.trace("Processing response");
+                            if (daily) {
+                                lastDailyResponse = response;
+                                pollForDataDailyData(lastDailyResponse);
+                            } else {
+                                lastHourlyResponse = response;
+                                pollForDataHourlyData(lastHourlyResponse);
+                            }
+                        });
+                        if (authFailedPreviously && getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                            updateStatus(ThingStatus.ONLINE);
+                        }
+                    }
+                } else {
+                    logger.debug("Failed to get latest MET office forecast");
+                    // Clear the web servers thread
+                    if (result != null) {
+                        final Response resp = result.getResponse();
+                        if (resp != null) {
+                            scheduler.execute(() -> {
+                                handleNon200Response(resp);
+                            });
+                        }
+                    }
+                }
+            }
+        };
+
+        sendAsyncSiteApiRequest(httpClient, daily, uplinkBridge.getApiKey(), location, listener);
+    }
+
+    protected static void sendAsyncSiteApiRequest(final HttpClient httpClient, final boolean daily, final String apiKey,
```

Comment:
I would prefer to move the http/api-handling to the bridge, as that is a better fit because that acts as a gateway to all (one or more) child `site` Things

Reply:
I've been doing quite a bit of refactoring after a lot of reflection to the comment :) so that the api layer is separated, and can be attached where required. Now the bridge is the manager for the API, with the child handlers using it via the bridge where required.